### PR TITLE
Divi/sign up

### PR DESCRIPTION
**Pull request by didvi**
[//]: # "These comments meant for your reference, they are invisible and don't need to be deleted"

## What's new in this PR?
Sign up page

### How to Review
`SignUp.js` is the main part
removed some junk in `utils.js`
made some changes to Myles' input and dropdown components (mostly just adding props)--will need to make sure they still work for his page and merge after his is merged

[//]: # 'The order in which to review files and 
what to expect when testing locally'
[//]: # '####### YOUR ANSWER BELOW ###########'
[//]: # '############## END ##################'

### Related PRs
Myles' PRs
[//]: # "Optional - related PRs you're waiting on
/ PRs that will conflict, etc"
[//]: # '####### YOUR ANSWER BELOW ###########'
[//]: # '############## END ##################'

### Design Status

[//]: # 'If this is a frontend PR, what is the expected 
status of the UI in this PR (lo, mid, high- fi?'
[//]: # '####### LOFI/MIDFI/HIFI ###########'
[//]: # '############## END ##################'

CC: @phoebeli23 @mylesdomingo

### CODE REVIEW

**Review by phoebeli23 — CHANGES_REQUESTED** [review]
- Clicking "Submit" to sign up does not work for me; generates the following error
Screen Shot 2021-04-08 at 12 57 07 PM

- Styling/prop error 
Screen Shot 2021-04-08 at 12 55 28 PM

File: src/screens/shared/auth/SignUp.js [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+    this.state = {
+      firstName: '',
+      lastName: '',
+      email: '',
+      password: '',
+      userTypes: '',
+      groupGapContact: false,
+      errors: {}
+    };
+    this.handleChange = this.handleChange.bind(this);
```
Comment (mylesdomingo) [inline src/screens/shared/auth/SignUp.js:22]: You can use arrow functions in a react class component to bind to `this` without calling bind. (eg. `handleChange = (name) => { ... }`)

File: src/screens/shared/auth/SignUp.js [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+        </form>
+        <br />
+        <br />
+        <div className="submit-button">
+          <Button
+            onClick={this.createAccount}
+            variant="contained"
+            color="inherit"
+            style={{
+              'background-color': 'var(--ks-dark-blue)',
```
Comment (mylesdomingo) [inline src/screens/shared/auth/SignUp.js:153]: change to `backgroundColor` 

use camelCase for MUI components

File: src/screens/shared/auth/SignUp.js [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+
+class SignUp extends React.PureComponent {
+  constructor(props) {
+    super(props);
+    this.state = {
+      firstName: '',
+      lastName: '',
+      email: '',
+      password: '',
+      userTypes: '',
```
Comment (mylesdomingo) [inline src/screens/shared/auth/SignUp.js:18]: `userTypesIndex` should be initialized at 1 or 0 or '', 
image

File: src/screens/shared/auth/SignUp.js [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+      <div className="sign-up-form">
+        <h1 id="sign-up-title">Create an Account</h1>
+        <form className="sign-up-card">
+          <h2>Contact Information</h2>
+          <div className="flex column">
+            <div className="row">
+              <FieldInput
+                label="First Name"
+                variant="outlined"
+                value={firstName}
```
Comment (mylesdomingo) [inline src/screens/shared/auth/SignUp.js:101]: i think we need to short circuit the values to fix the uncontrolled warning in the console. can you add this into the `FieldInput` code? 

example: `value={this.state.input || ""}`

File: src/screens/shared/auth/SignUp.js [hunk 5]
```diff
@@ -0,0 +1,166 @@
...
+      fieldsToValidate.map(f => validateField(f, state[f]))
+    ).catch(e => {
+      console.error(e);
+    });
+
+    const newErrors = {};
+    fieldsToValidate.forEach((field, i) => {
+      const errorMessage = allErrorMessages[i];
+      if (errorMessage !== '') {
+        newErrors[field] = errorMessage;
```
Comment (mylesdomingo) [inline src/screens/shared/auth/SignUp.js:67]: image

getting this error when i submit

**Review by mylesdomingo — CHANGES_REQUESTED** [review]

File: src/screens/shared/auth/SignUp.js (hunk 4 above)
Comment (mylesdomingo) [reply src/screens/shared/auth/SignUp.js:101]: https://stackoverflow.com/questions/47012169/a-component-is-changing-an-uncontrolled-input-of-type-text-to-be-controlled-erro

**Comment by didvi** [comment]
Resolved warnings in the console. User creation should work if you have this on your backend https://github.com/calblueprint/bp-boilerplate-backend/pull/9

**Comment by phoebeli23** [comment]
Things to address:
- See the screenshot from my earlier comment ("Unsupported style property background-color")
- Please center the component and move the submit button to the right ([refer to figma](https://www.figma.com/file/vWdNmoKEuSF24x9OYN9vJh/KS-NSEVP-Wireframes?node-id=2858%3A43041))

**Pull request merged**